swabbie-core/src/main/kotlin/com/netflix/spinnaker/swabbie/discovery/DiscoveryAware.kt

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2019 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.swabbie.discovery
+
+import com.netflix.spinnaker.kork.eureka.RemoteStatusChangedEvent
+import java.util.concurrent.atomic.AtomicBoolean
+
+open class DiscoveryAware : DiscoveryActivated {
```

Comment:
We have a few places where this is needed. Lifting this in a class so it can be used by any scheduler.

Reply:
What if we just added this up/down feature to `DiscoveryActivated`? I think that any class that is `DiscoveryActivated` wouldn't mind having this up atomic boolean, and that avoids another class to pick from.

Reply:
Fine by me, I suppose that can be implemented in `DiscoveryActivated`.